### PR TITLE
[KED-2957] Fix Metadata Panel for Modular Pipelines 

### PR DESCRIPTION
**Pull request by rashidakanchwala**
Signed-off-by: Rashida Kanchwala <rashida.kanchwala@quantumblack.com>

## Description

This PR was created based on the Jira Ticket - https://jira.quantumblack.com/browse/KED-2957 

## Development notes

This ticket handles two issues : - 
- If metadata panel is already open after clicking a node in the modular pipeline, then clicking on the modular pipeline in the side panel doesn't close it. This has been fixed, by closing metadata panel when modular pipeline is selected. Steps to produce this error : 
 --expand reporting pipelineline 
 -- click on any node in the pipeline 
 -- now click on reporting 
 
- Metadata panel used to read only visible node data. For modular pipelines, nodes that weren't expanded -- metadata panel wouldn't show anything. This has been fixed by changing metadata panel to read all nodes (expanded, collapsed) 

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: src/components/node-list/node-list.js [hunk 1]
```diff
@@ -25,6 +26,7 @@ const NodeList = ({
   onItemMouseLeave,
   onItemChange,
   onModularPipelineToggleExpanded,
+  onToggleNodeSelected,
```
Comment (tynandebold) [inline src/components/node-list/node-list.js:29]: Don't think you need this anymore.

**Review by tynandebold — APPROVED** [review]
Works just as described. Great work :)

File: src/components/node-list/node-list-tree.js [hunk 2]
```diff
@@ -190,16 +191,20 @@ const TreeListProvider = ({
...
       >
         {children}
       </NodeListTreeItem>
     );
   };
 
-  const onItemExpandToggle = (event, expandedItemIds) => {
+  const onItemExpandCollapseToggle = (event, expandedItemIds) => {
     onNodeToggleExpanded(expandedItemIds);
+    //when tree is collapsed
```
Comment (studioswong) [inline src/components/node-list/node-list-tree.js:204]:
```suggestion
    //when the parent modular pipeline tree of the selected node is collapsed
```

**Review by studioswong — COMMENTED** [review]
This overall fixes the problem brilliantly. My only comment is about the extra condition as I commented above, but other than than happy to approve once this is all clarified.

File: src/components/node-list/node-list-tree.js [hunk 3]
```diff
@@ -190,16 +191,20 @@ const TreeListProvider = ({
...
+    if (expandedItemIds.length === 0) {
+      onToggleNodeSelected(null);
+    }
```
[the remark this answers is not in the document]
Comment (rashidakanchwala) [reply src/components/node-list/node-list-tree.js:207]: @studioswong  -- this is for the up and down arrow/chevron. So expand the modular pipeline using the arrow/chevron down, click on a node it opens the metadata panel n now click back on the chevron/arrow so it collapses. U will notice the metadata panel doesn't close. So this is to close that.

**Review by studioswong — APPROVED** [review]
LGTM - great work ⭐

**Pull request merged**